biojava-structure/src/main/java/org/biojava/nbio/structure/symmetry/core/Stoichiometry.java

```diff
@@ -0,0 +1,290 @@
+	 * @return new {@link Stoichiometry} object.
+	 */
+	public Stoichiometry combineWith(Stoichiometry other) {
+		List<SubunitCluster> combinedClusters = new ArrayList<>();
+		combinedClusters.addAll(this.orderedClusters);
+		combinedClusters.addAll(other.orderedClusters);
+
+		// check that there is no alpha duplication
+		Set<String> thisAlpha = this.orderedClusters.stream().map(SubunitCluster::getAlpha).collect(Collectors.toSet());
+		Set<String> otherAlpha = other.orderedClusters.stream().map(SubunitCluster::getAlpha).collect(Collectors.toSet());
+		boolean resetAlphas = true;
+		if(Collections.disjoint(thisAlpha,otherAlpha)) {
+			resetAlphas = false;
+		}
+		Stoichiometry combinedStoichiometry = new Stoichiometry(combinedClusters,this.strategy,resetAlphas);
+
+		return combinedStoichiometry;
+	}
+
+	/**
+	 * Make a Stoichiometry object that corresponds to a single component.
+	 * @param i component index
+	 * @return new {@link Stoichiometry} object.
+	 */
+	public Stoichiometry getComponent(int i) {
+		return new Stoichiometry(Collections.singletonList(orderedClusters.get(i)),this.strategy,false);
+	}
+
+	/**
+	 * @return {@link StringOverflowStrategy} used in this stoichiometry
+	 *          to construct human-readable representation in case number
+	 *          of clusters exceeds number of letters in the alphabet.
+	 */
+	public StringOverflowStrategy getStrategy() {
+		return strategy;
+	}
+
+	/**
+	 * Change string representation of a stoichiometry in case number of clusters exceeds number of letters in the alphabet.
+	 * This action may invalidate alphas already assigned to the clusters.
+	 * @param strategy
+	 *          {@link StringOverflowStrategy} used in this stoichiometry
+	 *          to construct human-readable representation in case number
+	 *          of clusters exceeds number of letters in the alphabet.
+	 */
+	public void setStrategy(StringOverflowStrategy strategy) {
+		if(this.strategy != strategy) {
+			this.strategy = strategy;
+			if(orderedClusters.size()>alphabet.length())
+				doResetAlphas();
+		}
+	}
+
+	/**
+	 * @return Alphabet (a sequence of characters) used in this stoichiometry to construct human-readable representation.
+	 */
+	public String getAlphabet() {
+		return alphabet;
+	}
+
+	/**
+	 * Change alphabet used for string representation of a stoichiometry.
+	 * This action invalidates alphas already assigned to the clusters.
+	 * @param alphabet
+	 *          a sequence of characters used in this stoichiometry to construct human-readable representation.
+	 */
+	public void setAlphabet(String alphabet) {
+		this.alphabet = alphabet;
+		doResetAlphas();
+	}
+
+	/**
+	 * @return Human-readable representation of this stoichiometry.
+	 */
+	@Override
+	public String toString() {
+		String formula =
+				orderedClusters.stream().
+					map((SubunitCluster r)-> {
+						String output = r.getAlpha();
+						if(output.length()>1 || r.size()>1)
+							output+=r.size();
```

Comment:
This might be potentially slow for cases with many SubunitClusters. Ideally a StringBuilder should be used. See https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java